### DISCUSSION TITLE
Warning in React Native 0.65 => If you want to use Reanimated 2...

### DISCUSSION THREAD

**Discussion by Parterdev** [software-mansion/react-native-reanimated#2445] [Q&A]
First you must understand that the warning that appears is due to the fact that Reanimated 2 is in alpha phase, so there may be problems with the RN 0.65 version (current) and the Reanimated version (^ 2.) When installing it from React Navigation or with the command => $ yarn add react-native reanimated.

**To fix that warning:**

1. They should install the latest version of Reanimated with the command => $ yarn add react-native-reanimated@2.3.0-alpha.3 and add the following plugin integration in the babel.config.js of their projects.

```
module.exports = {
  presets: ['module: metro-react-native-babel-preset'],
  plugins: [
    'react-native-reanimated / plugin'
  ]
};
```

2.- You must activate Hermes (Javascript engine to optimize load time and memory consumption in our applications) in android / app / build.gradle:

```
project.ext.react = [
enableHermes: true, // clean and rebuild if changing
]
```

3.- Inject Reanimated in the MainApplication.java and create an override for the new JSIModulePackage.

```
import com.facebook.react.bridge.JSIModulePackage; // <- add
import com.swmansion.reanimated.ReanimatedJSIModulePackage; // <- add
```

After the getJSMainModuleName () method ...

```
@Override protected JSIModulePackage getJSIModulePackage () {
     return new ReanimatedJSIModulePackage (); // <- add
}
```

4.- Finally:

     $ cd yourApp / android /

     $ ./gradlew clean

     $ npx react-native run-android

**Comment by gianlucca** [comment]
thanks a lot!

I'm running v2.2.3 and did all other steps aside from installing alpha/beta version and it did the trick removing the warning.

**Comment by lmcjt37** [comment]
@Parterdev Any suggestions around fixing the same issue on iOS?

**Reply by matiasmm** [reply]
This does not solve the issue for iOS, only for android. We still need a solution about how to keep debugging mode active and use reanimated  2.

**Reply by ddimsoft** [reply]
> This does not solve the issue for iOS, only for android. We still need a solution about how to keep debugging mode active and use reanimated 2.

In my Case. 
This guide is useful. 

I don't add the line  'react-native-reanimated / plugin' in babel.config.js. 

Thanks for guide.

**Comment by saisindhupriya** [comment]
Thanks for guide i solved my warning message

**Comment by hokimtam** [comment]
> 

Save my life. thanks

**Comment by Stepan72** [comment]
Man, you are the best of the best! 😎

**Comment by MatiPl01** [comment]
I think that this discussion can be safely closed. It refers to reanimated v2, which is no longer maintained.

If still have questions, fell free to re-open this discussion or start a new one.
